tmt/steps/provision/__init__.py

```diff
@@ -222,14 +222,17 @@ def _ssh_options(self, join=False):
         if self.port:
             options.extend(['-p', str(self.port)])
         if self.key:
-            options.extend(['-i', self.key])
+            options.extend(['-i', f'"{self.key}"' if join else self.key])
         return ' '.join(options) if join else options
 
     def _ssh_command(self, join=False):
         """ Prepare an ssh command line for execution (list or joined) """
         command = ['sshpass', f'-p{self.password}'] if self.password else []
-        command += ['ssh'] + self._ssh_options()
-        return ' '.join(command) if join else command
+        command.append("ssh")
+        if join:
+            return " ".join(command) + " " + self._ssh_options(join=True)
```

Comment:
need to quote - self.password can contains single/double quotes

Reply:
Fixed using `shlex.quote` as well.